### PR TITLE
Scanner-Groups: Added search endpoint support 

### PR DESCRIPTION
**Pull request by sahils-tenable**
# Description

Added support for search endpoints and its tests

### CODE REVIEW

File: tenable/io/v3/vm/scanner_groups/api.py [hunk 1]
```diff
@@ -165,32 +170,102 @@ def edit(self, group_id: UUID, name: str) -> None:
...
+                    ...  }],
+                    ...  'and': [{
+                    ...      'value': '3',
+                    ...      'operator': 'oper',
+                    ...      'property': 3
+                    ...  }]
+                    ... }
+                As the filters may change and sortable fields may change over
+                time, it's highly recommended that you look at the output of
+                endpoint to get more details.
```
Comment (harshshah-tenable) [inline tenable/io/v3/vm/scanner_groups/api.py:215]: Please add the respective filter/definition method in docstring.

**Review by milanaghera-tenable — APPROVED** [review]

File: tenable/io/v3/vm/scanner_groups/api.py [hunk 2]
```diff
@@ -134,15 +137,15 @@ def details(self, group_id: UUID) -> Dict:
             group_id (UUID): The unique identifier for the scanner group.
 
         Returns:
-            :obj:`dict`:
```
Comment (ngaglani-tenable) [inline tenable/io/v3/vm/scanner_groups/api.py:137]: Should we revert this to `dict`?

Comment (sahils-tenable) [reply tenable/io/v3/vm/scanner_groups/api.py:137]: we will do this at final stage for all endpoints

File: tenable/io/v3/vm/scanner_groups/api.py [hunk 3]
```diff
@@ -165,32 +170,103 @@ def edit(self, group_id: UUID, name: str) -> None:
...
+                specific page of data.
+            return_csv (bool, optional):
+                If set to true, it will return the CSV response or
+                iterable (based on return_resp flag). Iterator returns all
+                rows in text/csv format for each call with row headers.
+        Returns:
+            Iterable:
+                The iterable that handles the pagination for the job.
+            requests.Response:
+                If ``return_json`` was set to ``True``, then a response
```
Comment (alpesh-te) [inline tenable/io/v3/vm/scanner_groups/api.py]: Update docstring.

File: tests/io/v3/vm/scanner_groups/test_scanner_group_api.py [hunk 4]
```diff
@@ -181,3 +184,73 @@ def test_edit_routes(api):
...
+
+    iterator = api.v3.vm.scanner_groups.search(
+        fields=fields, limit=200, sort=sort, filter=filters
+    )
+    assert isinstance(iterator, SearchIterator)
+
+    event_list = []
+    for event in iterator:
+        event_list.append(event)
+    assert len(event_list) == api_response['pagination']['total']
```
Comment (alpesh-te) [inline tests/io/v3/vm/scanner_groups/test_scanner_group_api.py]: get rid of loop.

**Review by alpesh-te — CHANGES_REQUESTED** [review]
Minor fixes!

File: tenable/io/v3/vm/scanner_groups/api.py (hunk 3 above)
Comment (alpesh-te) [reply tenable/io/v3/vm/scanner_groups/api.py]: Addressed

File: tests/io/v3/vm/scanner_groups/test_scanner_group_api.py (hunk 4 above)
Comment (alpesh-te) [reply tests/io/v3/vm/scanner_groups/test_scanner_group_api.py]: Addressed

**Pull request merged**